api/base/serializers.py

```diff
@@ -88,7 +89,34 @@ def to_internal_value(self, data):
         return super(JSONAPIListField, self).to_internal_value(data)
 
 
-class JSONAPIHyperlinkedIdentityField(ser.HyperlinkedIdentityField):
+class AuthorizedCharField(ser.CharField):
+    """
+    Passes auth of the logged-in user to the object's method
+    defined as the field source.
+
+    Example:
+        content = AuthorizedCharField(source='return_content')
+    """
+    def __init__(self, source=None, **kwargs):
+        assert source is not None, 'The `source` argument is required.'
+        self.source = source
+        super(AuthorizedCharField, self).__init__(source=self.source, **kwargs)
+
+    def get_attribute(self, obj):
+        user = self.context['request'].user
+        auth = auth_core.Auth(user)
+        field_source_method = getattr(obj, self.source)
+        return field_source_method(auth=auth)
+
+
+class HyperlinkedFieldMixin(object):
```

Comment:
The design goal here was to handle a family of related hyperlinked field types that all had to be represented a specific way: 
https://github.com/CenterForOpenScience/osf.io/pull/4419/files#diff-4b396d38a0947821679352d308445c5aR415

Previously we were using `isinstance` checks that called out each of 3+ field classes. (which seemed fragile to maintain) As the person who originally suggested a mixin, I'm 100% open to alternative solutions for that part.